vllm/entrypoints/openai/api_server.py

```diff
@@ -45,18 +45,15 @@ async def _force_log():
 
 
 app = fastapi.FastAPI(lifespan=lifespan)
+# Instrument the app with HTTP metrics and expose it on /metrics
+Instrumentator().instrument(app).expose(app, endpoint="/metrics")
 
 
 def parse_args():
     parser = make_arg_parser()
     return parser.parse_args()
 
 
-# Add prometheus asgi middleware to route /metrics requests
-metrics_app = make_asgi_app()
-app.mount("/metrics", metrics_app)
-
-
```

Comment:
This is how the metrics defined in `vllm/engine/metrics.py` are exposed. It can't be removed.

Reply:
I replaced it with the `expose()` method of `prometheus-fastapi-instrumentator` which also exposes a `/metric` endpoint.
https://github.com/vllm-project/vllm/pull/2764/files#diff-38318677b76349044192bf70161371c88fb2818b85279d8fc7f2c041d83a9544R48-R49

I noticed it also solves the `/metrics/` redirection [issue](https://github.com/prometheus/client_python/issues/1016).
Which of the exposing methods should we use?

Reply:
> I replaced it with the expose() method of prometheus-fastapi-instrumentator which also exposes a /metric endpoint.

While this does expose a `/metrics` endpoint, none of the vLLM metrics will be in it because they come from `make_asgi_app()`, right?

Have you confirmed that `/metrics` still contains vLLM metrics with this code removed?

Reply:
"Yes, I've verified that both approaches expose all metrics. The only discrepancy I've noticed is that `expose()` from `prometheus-fastapi-instrumentator` exposes metrics on `/metrics`, whereas `make_asgi_app()` exposes them on `/metrtics/`. However, I'll revert to using the `make_asgi_app()` approach. I find the other method somewhat hacky, as it involves the `prometheus-fastapi-instrumentator` middleware handling the metrics endpoint. This could look weird if multiple middlewares are in use."